host/src/db.rs

```
@@ -109,20 +91,61 @@ pub async fn store_blob_proof(
 pub async fn retrieve_blob_id_proof(
     db_pool: Arc<Mutex<Pool<Postgres>>>,
     blob_id: String,
-) -> Option<String> {
+) -> Option<(Option<String>, bool)> {
     let db_lock = db_pool.lock().await;
 
-    let result = sqlx::query(
+    sqlx::query(
         r#"
-            SELECT PROOF FROM BLOB_PROOFS
+            SELECT PROOF, FAILED FROM BLOB_PROOFS
             WHERE BLOB_ID = $1
             "#,
     )
     .bind(blob_id)
     .fetch_optional(&*db_lock)
     .await
     .ok()?
-    .map(|row| row.get::<Option<String>, _>("proof"))?;
+    .map(|row| {
+        let proof: Option<String> = row.get("proof");
+        let failed: bool = row.get("failed");
+        (proof, failed)
+    })
+}
+
```

When trying to retrieve a proof that failed, this is returning `None` instead of `Some(None,true)`